### PR TITLE
Modularize DDD functionality

### LINKED ISSUE #2562: Modularize DDD
Similar to the work done in #2452, we want to create an API where a user has access to intermediary circuits and can control the inference process.

We also want to update the documentation such that the two-stage API functions are demonstrated in the user guide.

### PR DESCRIPTION
**Pull request by bdg221**
## Description
---
This PR introduces two functions

- `generate_circuits_with_ddd`
- `generate_ddd_value`

for generating circuits with DDD sequences and using the results from running those circuits to get an expectation value. The first function moves generating the circuits outside of `execute_with_ddd`, allowing users to generate and return DDD circuits.

The second function takes the results from all of the DDD circuits used in `execute_with_ddd` and averages the results to produce a single expectation value (float) that is returned.

fixes #2562

### CODE REVIEW

**Comment by bdg221** [comment]
@natestemen Do you think the second function `generate_ddd_value` is necessary since it is just calling `np.average`?

**Comment by bdg221** [comment]
On the community call, it was agreed that the second function is important for consistency with ZNE and PEC. Also, I changed `generate_ddd_value` to `combine_results` for consistency.

File: mitiq/ddd/tests/test_ddd.py [hunk 1]
```diff
@@ -226,3 +231,14 @@ def exec_xx_small_spacing(circuit):
     # What is important to test is getting different results.
     assert not np.isclose(unmitigated, mitigated_small_spacing)
     assert not np.isclose(mitigated_large_spacing, mitigated_small_spacing)
+
+
+@mark.parametrize("num_trials", [1, 3, 5])
```
Comment (natestemen) [inline mitiq/ddd/tests/test_ddd.py:236]: Since this test should be fast (correct me if i'm wrong) can we add a few more trials? Say up to 30?

**Review by natestemen — APPROVED** [review]
LGTM! Left one comment, and my only other hold-up was thinking `generate_circuits_with_ddd` is a bit of a mouthful, but I can't think of a better name!

**Comment by natestemen** [comment]
Going to merge before CI passes. I ran the new test locally and it looks good!

**Pull request merged**